internal/isoeditor/rhcos.go

```diff
@@ -382,6 +404,66 @@ func IgnitionImageArchive(ignitionConfig string) ([]byte, error) {
 	return compressedBuffer.Bytes(), nil
 }
 
+func readHeader(isoPath string) (*OffsetInfo, *OffsetInfo, error) {
+	iso, err := os.OpenFile(isoPath, os.O_RDONLY, 0o664)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer iso.Close()
+
+	// Read header
+	header := make([]byte, headerLength)
+	_, err = iso.Read(header)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var offsetInfos []*OffsetInfo
+
+	// Read OffsetInfos
+	headerBytes := header[len(header)-24:]
+	offsetInfo, err := readOffsetInfo(headerBytes)
+	if err != nil {
+		return nil, nil, err
+	}
+	offsetInfos = append(offsetInfos, offsetInfo)
+
+	headerBytes = header[len(header)-48 : len(header)-24]
+	offsetInfo, err = readOffsetInfo(headerBytes)
+	if err != nil {
+		return nil, nil, err
+	}
+	offsetInfos = append(offsetInfos, offsetInfo)
+
+	var ignitionOffsetInfo *OffsetInfo
+	var ramdiskOffsetInfo *OffsetInfo
+	for _, o := range offsetInfos {
+		key := string(o.Key[:])
+		switch key {
+		case ignitionHeaderKey:
+			ignitionOffsetInfo = o
+		case ramdiskHeaderKey:
+			ramdiskOffsetInfo = o
+		default:
+			// Invalid key
+			return nil, nil, errors.Errorf("Invalid key in ISO header: %s", key)
+		}
+	}
+
+	return ignitionOffsetInfo, ramdiskOffsetInfo, nil
+}
+
+func readOffsetInfo(headerBytes []byte) (*OffsetInfo, error) {
```

Comment:
Maybe `parseOffsetInfo`?
We've already read the info from the file.

Reply:
yeah, added 'ParseOffsetInfo(headerBytes []byte, key string)' instead.